mirrord-agent/src/runtime.rs

```diff
@@ -1,68 +1,65 @@
 use std::{
     fs::File,
     os::unix::io::{IntoRawFd, RawFd},
+    path::PathBuf,
 };
 
 use anyhow::{anyhow, Result};
+use bollard::{container::InspectContainerOptions, Docker};
 use containerd_client::{
     connect,
-    services::v1::{containers_client::ContainersClient, GetContainerRequest},
+    services::v1::{tasks_client::TasksClient, GetRequest},
     tonic::Request,
     with_namespace,
 };
 use nix::sched::setns;
-use serde::{Deserialize, Serialize};
+pub(crate) enum Runtime {}
 
-const CONTAINERD_SOCK_PATH: &str = "/run/containerd/containerd.sock";
-const DEFAULT_CONTAINERD_NAMESPACE: &str = "k8s.io";
-
-#[derive(Serialize, Deserialize, Debug)]
-struct Namespace {
-    #[serde(rename = "type")]
-    ns_type: String,
-    path: Option<String>,
-}
-
-#[derive(Serialize, Deserialize, Debug)]
-struct LinuxInfo {
-    namespaces: Vec<Namespace>,
-}
-
-#[derive(Serialize, Deserialize, Debug)]
-struct Spec {
-    linux: LinuxInfo,
+impl Runtime {
+    pub async fn get_container_pid(container_id: &str, container_runtime: &str) -> Result<PathBuf> {
+        match container_runtime {
```

Comment:
`match`ing on strings like this feels weird, maybe you could have an enum that wraps the string + container_id, like:

```suggestion
        match container_runtime {
            Runtime::Docker(container_id) => ...
            Runtime::Containerd(container_id) => ...
```